arch/riscv/core/smp.c

```diff
@@ -27,6 +32,15 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	riscv_cpu_sp = Z_KERNEL_STACK_BUFFER(stack) + sz;
 	riscv_cpu_wake_flag = _kernel.cpus[cpu_num].arch.hartid;
 
+#ifdef CONFIG_PM_CPU_OPS
+	uintptr_t entry = (uintptr_t)&__start;
+
+	if (pm_cpu_on(cpu_num, entry)) {
```

Comment:
I don't get it. Is RISCV supporting PSCI and SMCCC now? What PM CPU ops driver is this using?

Reply:
I am not sure why PM CPU ops tightly coupled with PSCI or SMCC.  RISCV SMP's arch_start_cpu() assumes secondary core is already powered up and looping at boot_secondary_core, but some platforms needs to do initialization (either via opensbi or regular MMIO to enable clocks, etc,.) to bring up secondary cores.

As of now, RISCV doesn't have any PM CPU Ops driver, but platforms can implement weak functions to bringup secondary cores.

Reply:
My point is exactly that RISCV doesn't have any PM CPU ops driver in tree so you are adding this for some out-of-tree platform?

Reply:
Right, as of now the platform is out of tree.

Reply:
ok, do you plan to upstream it? I'm not willing to push code into the tree that is basically dead code without any user for it.

Reply:
Summarizing the offline conversation we had...

This code is in support of a microcontroller core embedded in a larger SoC design so there's not a clear path to share a buildable platform.  In this particular platform the driver implementing the CPU Ops talks to some proprietary power/clock/reset controlling hardware.

As a separate discussion supporting Zephyr-on-SBI on an upstream platform (at least QEMU) would be really interesting.